drivers/can/can_mcp2515.c

```diff
@@ -271,6 +271,18 @@ const int mcp2515_set_mode(struct device *dev, uint8_t mcp2515_mode)
 	return 0;
 }
 
+static int mcp2515_get_mode(struct device *dev)
+{
+	uint8_t canstat;
+
+	if (mcp2515_cmd_read_reg(dev, MCP2515_ADDR_CANSTAT, &canstat, 1)) {
+		return -EIO;
+	}
+
+	return (canstat & MCP2515_CANSTAT_MODE_MASK)
```

Comment:
Not sure about this, but are we allowed to return both data and return codes in functions at the same time? I believe it'd be better to return 0 on success and the mode in a reference argument. Should be inlined by the compiler anyways with a higher optimization mode.